### PR TITLE
CI: Add Ruby 3.1

### PR DESCRIPTION
**Pull request by ashie**
Signed-off-by: Takuro Ashie <ashie@clear-code.com>

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -30,3 +32,17 @@ environment:
...
+  matrix:
+    only:
+      - ruby_version: "31-x64"
+  install:
+    - SET PATH=C:\Ruby%ruby_version%\bin;%PATH%
+    - ruby --version
+    - gem --version
+    - bundle --version
+    - ridk.cmd install 1 3
+    - rixk.cmd bundle install
```
Comment (cosmo0920) [inline appveyor.yml:47]: typo?

Comment (ashie) [reply appveyor.yml]: Fixed.

**Review by cosmo0920 — COMMENTED** [review]
Looks good but shouldn't we add Ruby 3.1 task on GitHub Actions workflow here?
https://github.com/fluent-plugins-nursery/winevt_c/blob/master/.github/workflows/linux.yml

**Review by cosmo0920 — APPROVED** [review]

**Comment by ashie** [comment]
> Looks good but shouldn't we add Ruby 3.1 task on GitHub Actions workflow here? https://github.com/fluent-plugins-nursery/winevt_c/blob/master/.github/workflows/linux.yml

My first commit in this PR was just adding it, but I temporarily removed the commit because I noticed a weird thing.
It uses rake-compiler-dock to run cross-compile, so it seems meaningless to add host OS's ruby version.
Adding Ruby 3.1 on x64-mingw-ucrt here seems desired:
https://github.com/fluent-plugins-nursery/winevt_c/blob/144ef8f291e9038a0c0dc876e4f08c3c2568e8fd/Rakefile#L30-L33
but I don't yet get succeeded to build.

Since we don't provide fat gem and we already confirmed it's buildable on Windows, I may abandon investigating it.

**Comment by ashie** [comment]
The patch I tried:
```diff
diff --git a/Rakefile b/Rakefile
index 7b61378..833de94 100644
--- a/Rakefile
+++ b/Rakefile
@@ -20,7 +20,7 @@ Rake::ExtensionTask.new("winevt", spec) do |ext|
   ext.cross_compile = true
   ext.lib_dir = File.join(*['lib', 'winevt', ENV['FAT_DIR']].compact)
   # cross_platform names are of MRI's platform name
-  ext.cross_platform = ['x86-mingw32', 'x64-mingw32']
+  ext.cross_platform = ['x86-mingw32', 'x64-mingw32', 'x64-mingw-ucrt']
 end
 
 desc 'Build gems for Windows per rake-compiler-dock'
@@ -30,6 +30,7 @@ task 'gem:native' do
   %w[ x86-mingw32 x64-mingw32 ].each do |plat|
     RakeCompilerDock.sh "bundle --local && bundle exec rake native:#{plat} gem RUBY_CC_VERSION=2.4.0:2.5.0:2.6.0:2.7.0:3.0.0", platform: plat
   end
+  RakeCompilerDock.sh "bundle --local && bundle exec rake native:x64-mingw-ucrt  gem RUBY_CC_VERSION=3.1.0", platform: "x64-mingw-ucrt"
 end
 
 CLEAN.include('lib/winevt/winevt.*')
diff --git a/winevt_c.gemspec b/winevt_c.gemspec
index b5c115b..4c953cb 100644
--- a/winevt_c.gemspec
+++ b/winevt_c.gemspec
@@ -28,7 +28,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "bundler", [">= 1.16", "< 3"]
   spec.add_development_dependency "rake", "~> 13.0"
   spec.add_development_dependency "rake-compiler", "~> 1.0"
-  spec.add_development_dependency "rake-compiler-dock", "~> 1.1.0"
+  spec.add_development_dependency "rake-compiler-dock", "~> 1.2.2"
   spec.add_development_dependency "test-unit", "~> 3.2"
   spec.add_development_dependency "yard", "~> 0.9"
 end
```

and the result:
```
cd tmp/x86_64-linux/winevt/3.1.0
/usr/local/rvm/rubies/ruby-3.1.0/bin/ruby -I. -r.rake-compiler-siteconf.rb ../../../../ext/winevt/extconf.rb
Calling `DidYouMean::SPELL_CHECKERS.merge!(error_name => spell_checker)' has been deprecated. Please call `DidYouMean.correct_error(error_name, spell_checker)' instead.
checking for -lwevtapi... no
checking for EvtQuery() in winevt.h... no
checking for -ladvapi32... no
checking for -lole32... no
creating Makefile
cd -
cd tmp/x86_64-linux/winevt/3.1.0
/usr/bin/make
compiling ../../../../ext/winevt/winevt.c
In file included from ../../../../ext/winevt/winevt.c:1:
../../../../ext/winevt/winevt_c.h:8:10: fatal error: w32api.h: No such file or directory
    8 | #include <w32api.h>
      |          ^~~~~~~~~~
compilation terminated.
make: *** [Makefile:246: winevt.o] Error 1
rake aborted!
```

**Comment by cosmo0920** [comment]
Ah, yes. GitHub Actions is only used for testing to build fat gem.
Abandoning to add Ruby 3.1 on GitHub Actions is reasonable.

**Pull request merged**